stl/inc/chrono

```diff
@@ -2209,6 +2221,2946 @@ namespace chrono {
+            // seconds {100, 200, 300, 400}, we have:
+            //
+            //  UTC   sys   *_It
+            //   99    99    100
+            //  100     X    200
+            //  101   100    200
+            //  102   101    200
+            //  ...
+            //  199   198    200
+            //  200   199    300^
+            //  201     X    300
+            //  202   200    300
+            //  ...
+            //  299   297    300
+            //  300   298    400^
+            //  301   299    400^
+            //  302     X    400
+            //  303   300    400
+            //
+            //  ^_It points to 2nd leap second
+
+            _It = _STD upper_bound(_Ls_vector.begin(), _Ls_vector.end(), sys_seconds{_Time_floor.time_since_epoch()});
+        } else {
+            seconds _Prev_elapsed{0};
+            for (_It = _Ls_vector.begin(); _It != _Ls_vector.end(); ++_It) {
+                // UTC time when leap second insertion begins. In all cases, _It->date() + _It->_Elapsed() is the *end*
+                // of the insertion. For a negative leap that's also the beginning, but for a positive one, insertion
+                // begins 1 second earlier.
+                const utc_seconds _This_ls_begin{
+                    _It->date().time_since_epoch() + (_It->_Positive() ? _Prev_elapsed : _It->_Elapsed())};
+                if (_This_ls_begin > _Time_floor) {
+                    break;
+                }
+                _Prev_elapsed = _It->_Elapsed();
+            }
+        }
+
+        if (_It == _Ls_vector.begin()) {
+            return {false, seconds{0}};
+        } else {
+            // Convert to the last leap second before or equal to _Time.
+            const auto& _Last_leap = *--_It;
+            const utc_seconds _Utc_leap_second{_Last_leap.date().time_since_epoch() + _It->_Elapsed() - seconds{1}};
+#ifdef __cpp_lib_concepts
+            const auto _Leap_cmp = _Utc_leap_second <=> _Time_floor;
+#else // ^^^ __cpp_lib_concepts / TRANSITION, GH-395 workaround vvv
+            const auto _Leap_cmp = _Utc_leap_second > _Time_floor  ? strong_ordering::greater
+                                 : _Utc_leap_second == _Time_floor ? strong_ordering::equal
+                                                                   : strong_ordering::less;
+#endif // ^^^ workaround
+            if (_Tzdb._All_ls_positive && _STD is_gt(_Leap_cmp)) { // Case (1)
+                --_It;
+            }
+            return {_Last_leap._Positive() && _STD is_eq(_Leap_cmp), _It->_Elapsed()};
+        }
+    }
+
+    // CLASS utc_clock
+    class utc_clock {
+    public:
+        using rep                       = system_clock::rep;
+        using period                    = system_clock::period;
+        using duration                  = duration<rep, period>;
+        using time_point                = time_point<utc_clock>;

```

Comment:
`system_clock` uses `_CHRONO` qualification for `duration<` and `time_point<` here, although it's technically unnecessary, because this almost-shadowing is somewhat confusing. (The Standard also depicts `chrono::` qualified names.) `utc_clock` and `tai_clock` should be consistent. `gps_clock` is qualifying with `chrono::` instead of `_CHRONO`.